packages/protocol/contracts/common/MetaTransactionWallet.sol

```diff
@@ -213,8 +213,16 @@ contract MetaTransactionWallet is
     );
     uint256 dataPosition = 0;
     for (uint256 i = 0; i < destinations.length; i++) {
-      executeTransaction(destinations[i], values[i], sliceData(data, dataPosition, dataLengths[i]));
-      dataPosition = dataPosition.add(dataLengths[i]);
+      require(
+        uint8(data[dataPosition]) == dataLengths[i],
+        "Data length must match specified length"
+      );
+      executeTransaction(
+        destinations[i],
+        values[i],
+        sliceData(data, dataPosition.add(1), dataLengths[i])
+      );
+      dataPosition = dataPosition.add(dataLengths[i].add(1));
```

Comment:
ditto `dataLengths + 1`?

Reply:
there's a byte that now accounts for the length of the following byte segment. So when we move the cursor down, we now have to account for this extra length bytes "metadata".

Reply:
NOT backwards compatible though. Does that cause any problems?